### PR TITLE
Generate sdk for Microsoft.Features unregister api

### PR DESCRIPTION
**Pull request by diwudd**
Generate sdk for Microsoft.Features unregister api

### CODE REVIEW

File: sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/OperationDisplay.cs [hunk 1]
```diff
@@ -29,19 +29,16 @@ public OperationDisplay()
...
-        /// <param name="provider">Service provider:
-        /// Microsoft.Resources</param>
+        /// <param name="provider">Service provider: Microsoft.Features</param>
         /// <param name="resource">Resource on which the operation is
         /// performed: Profile, endpoint, etc.</param>
         /// <param name="operation">Operation type: Read, write, delete,
         /// etc.</param>
-        /// <param name="description">Description of the operation.</param>
-        public OperationDisplay(string provider = default(string), string resource = default(string), string operation = default(string), string description = default(string))
+        public OperationDisplay(string provider = default(string), string resource = default(string), string operation = default(string))
```
Comment (isra-fel) [inline sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/OperationDisplay.cs:37]: I wonder if this breaking change is intentional?

**Comment by isra-fel** [comment]
Could you please also provide a link to your openapi spec PR?

**Comment by diwudd** [comment]
Thank you for the review!

This is the PR link. https://github.com/Azure/azure-rest-api-specs/pull/8788

Thanks,

Di

**Comment by isra-fel** [comment]
"OperationDisplay" has multiple definitions in the openapi spec of resources, for example in [subscriptions.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json#L529) and [features.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Features/stable/2015-12-01/features.json#L330), but the one in features misses a property "description", so this PR introduced a breaking change.

Please fix it in the spec then generate again.
Thanks

**Comment by diwudd** [comment]
Hi Yeming,

Thank you for the review.

I am not sure I get it. The features.json did have the “description<https://github.com/Azure/azure-rest-api-specs/blob/3e8fb815e2e4f149fbfcdaab64efc4d45dcc0690/specification/resources/resource-manager/Microsoft.Features/stable/2015-12-01/features.json#L331>” property. So what is being missed here?

Thanks,

Di

**Comment by isra-fel** [comment]
OK, in the last SDK, the class `OperationDisplay` was generated from `subscriptions.json`, so it had a property named `Description`. Now in `features.json` there's another operationdisplay. Because the two share the same name, the model generated from features will replace the one from subscriptions. And since it doesn't have a description, from .NET SDK's perspective it's a breaking change that a property is deleted from one class.

**Comment by diwudd** [comment]
Hi Yeming,

Sorry, I am not sure I get it. The description did exist, anything else you are expecting?

"Operation": {
      "description": "Microsoft.Features operation",
      "type": "object",
      "properties": {
        "name": {
          "description": "Operation name: {provider}/{resource}/{operation}",
          "type": "string"
        },
        "display": {
          "description": "The object that represents the operation.",
          "properties": {
            "provider": {
              "description": "Service provider: Microsoft.Features",
              "type": "string"
            },
            "resource": {
              "description": "Resource on which the operation is performed: Profile, endpoint, etc.",
              "type": "string"
            },
            "operation": {
              "description": "Operation type: Read, write, delete, etc.",
              "type": "string"
            }
          }
        }
      }
    },
    "OperationListResult": {
      "description": "Result of the request to list Microsoft.Features operations. It contains a list of operations and a URL link to get the next set of results.",
      "properties": {
        "value": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Operation"
          },
          "description": "List of Microsoft.Features operations."
        },
        "nextLink": {
          "type": "string",
          "description": "URL to get the next set of operation list results if there are any."
        }
      }
    }
  },


Thanks,

Di

**Comment by isra-fel** [comment]
Yes it's a little bit ambiguous. I mean the "description" as a **property** of "display", not the description of "display". As I said [subscriptions.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json#L529) has a "description" property, but [features.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Features/stable/2015-12-01/features.json#L330) doesn't.

**Comment by isra-fel** [comment]
Hi @diwudd  any updates?

**Comment by diwudd** [comment]
Thank you, Yeming.

Now I got it, sent out a PR<https://github.com/Azure/azure-rest-api-specs/pull/9677> to fix it.

Thanks,

Di

**Comment by diwudd** [comment]
Ok, after adding the property, my Swagger failed validation as it added a new property. It is a breaking change so that I need to bump up the version.

But given that, how can the feature.json affect subscriptions.json? As they are different files, how can it affect each other and make it a breaking change in .NET sdk? Isn’t it a bug in the code generation tool?

Thanks,

Di

**Comment by isra-fel** [comment]
I checked your swagger PR, the breaking change was not related to any of your updates. You were merely adding a new operation and a new attribute after all.

> As they are different files, how can it affect each other and make it a breaking change in .NET sdk?

They are different files in swagger, but the SDKs generated from either are put / combined in the same directory / namespace. So if a model is defined in both files, its definitions must be exactly the same, otherwise one of features' client or subscription's will not get the model it expects, which is just the problem we have.

**Comment by diwudd** [comment]
Thanks Yeming.

I have fixed the issue. Can you take another look a the PR?

Thanks,

Di

**Comment by isra-fel** [comment]
The changes look good to me. Could you resolve the conflits to enable CI? Thanks

**Comment by diwudd** [comment]
Thank you Yeming. The conflicts have been resolved.

Thanks,

Di

**Comment by diwudd** [comment]
+@Ife<mailto:Ifeoluwa.Okunoren@microsoft.com>

Thanks,

Di

From: Di Wu
Sent: Monday, June 15, 2020 2:51 PM
To: Azure/azure-sdk-for-net <reply@reply.github.com>; Azure/azure-sdk-for-net <azure-sdk-for-net@noreply.github.com>
Cc: Mention <mention@noreply.github.com>
Subject: RE: [Azure/azure-sdk-for-net] Generate sdk for Microsoft.Features unregister api (#11721)

Thank you Yeming. The conflicts have been resolved.

Thanks,

Di

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**